src/constants.ts

```diff
@@ -0,0 +1,42 @@
+import { v4 } from 'uuid';
+import { ComparatorType } from './comparator';
+import { InterfaceOptionalKeys, noop } from './utils';
+import { maxComparator, noSortComparator } from './comparator';
+
+export const ADD_MESSAGE = '@@flashr/__ADD_MESSAGE__';
+export const CLEAR_MESSAGE = '@@flashr/__CLEAR_MESSAGE__';
+export const CLEAR_ALL_MESSAGES = '@@flashr/__CLEAR_ALL_MESSAGES__';
+export const NO_OP_ACTION = '@@flashr/__NO_OP_ACTION__';
+
+export interface ConfigInterface {
+  timeout: number;
+  position:
+    | 'left-top'
+    | 'center-top'
+    | 'right-top'
+    | 'left-bottom'
+    | 'center-bottom'
+    | 'right-bottom';
+  stackCount: number;
+  sortQueue: boolean;
+  comparator: ComparatorType;
+  keyFunction: () => string;
+  onActionClick: () => void;
+  onClick: () => void;
+}
+
+export function withDefaultsConfig(
+  config: InterfaceOptionalKeys<ConfigInterface>
```

Comment:
You could use the built in TS type here `Partial`

https://www.typescriptlang.org/docs/handbook/utility-types.html#partialt